src/legacy/core_plugins/data/public/search/aggs/buckets/date_histogram.ts

```diff
@@ -91,24 +95,36 @@ export const dateHistogramBucketAgg = new BucketAggType<IBucketDateHistogramAggC
   },
   createFilter: createFilterDateHistogram,
   decorateAggConfig() {
+    const uiSettings = getUiSettings();
     let buckets: any;
+
     return {
       buckets: {
         configurable: true,
         get() {
           if (buckets) return buckets;
 
-          buckets = new TimeBuckets();
-          buckets.setInterval(getInterval(this));
-          setBounds(this);
+          buckets = new TimeBuckets({ uiSettings });
+          updateTimeBuckets(this, buckets);
```

Comment:
i think we should be able to skip this call to updateTimeBuckets as well (and then it doens't need the second argument)

Reply:
interestingly i needed to keep this in place too in order for that data table functional test to pass. but i have a feeling we may still be able to get rid of this eventually.